src/documents/ContextAwareness.md

```diff
@@ -0,0 +1,327 @@
+---
+title: ContextAwareness API
+layout: default
+category: API
+---
+
+The ContextAwareness API of Infusion provides a powerful suite of grades and utilities that allow a component to be responsive to aspects of its contexts in a flexible and open-ended way.
+For lightweight cases of context adaptation, the [`distributeOptions`](IoCSS.md) scheme suffices, in that it enables a component to be responsive to broadcasts "advising" it from around the
+component tree in a free-form and controllable away. However, when a component needs to take a clearer role in responding to potentially many kinds of requests for adaptation, where these
+can be organised around two or more "axes" or "dimensions" of adaptation that can be clearly named and identified, it should opt into this more heavywight scheme by implementing the 
+`fluid.contextAware` grade and perhaps also advertising some of its dimensions of adaptations in its own options. These dimensions themselves can always be extended by further contributions
+to the component's options through all the usual sources - direct options, subcomponent options, and options distributions from elsewhere in the tree.
+
+A `contextAware` component can be responsive to contexts, whose influence is organised as a set of mutually orthogonal "adaptations", where the set of adaptations itself is open to extension by 
+integrators and adopters. The primary grade enabling a component to opt into this scheme is named `fluid.contextAware`. Any component derived from this grade will advertise an
+area of its options named `contextAwareness` which organises the rules for its adaptation in a hierarchical way - at the top level by "adaptation" and then at the nested level by
+"checks".
+
+## Structure of members in a `adaptationRecord`
```

Comment:
I think in future we will want to structure our "schematic" documentation around "named types of things" which can occur in various particular places in configuration - and also be automatically checked by our authoring tools. As such, we want "adaptationRecord" to be a real thing rather than merely a conceptual thing, and this represents the start of it...